lib/src/authentication/src/client_options.dart

```diff
@@ -1,192 +1,157 @@
 import 'package:ably_flutter/ably_flutter.dart';
 
-/// Ably library options used when instancing a REST or Realtime client library
-///
-/// https://docs.ably.com/client-lib-development-guide/features/#TO1
+/// Passes additional client-specific properties to the REST
+/// [Rest.new] or the Realtime [Realtime.new]
 class ClientOptions extends AuthOptions {
-  /// Optional clientId that can be used to specify the identity for this client
-  ///
-  /// In most cases it is preferable to instead specific a clientId in the token
-  /// issued to this client.
-  /// https://docs.ably.com/client-lib-development-guide/features/#TO3a
+  /// A client ID, used for identifying this client when publishing messages or
+  /// for presence purposes. The `clientId` can be any non-empty string, except
+  /// it cannot contain a `*`. This option is primarily intended to be used in
+  /// situations where the library is instantiated with a key. Note that a
+  /// `clientId` may also be implicit in a token used to instantiate the
+  /// library. An error will be raised if a `clientId` specified here conflicts
+  /// with the `clientId` implicit in the token.
   String? clientId;
 
-  /// Custom log handler
-  ///
-  /// For discussion about removing this component see
-  /// https://github.com/ably/ably-flutter/issues/238
-  ///
-  /// https://docs.ably.com/client-lib-development-guide/features/#TO3c
+  /// Controls the log output of the library. This is a function to handle each
+  /// line of log output.
   @Deprecated(
     'Not used, as log messages are handled by the default mechanism '
     'in the underlying SDK. This instance variable will be removed '
     'in a future release.',
   )
   LogHandler? logHandler;
 
-  /// Controls the level of verbosity of log messages from the library
-  ///
-  /// Use constants from [LogLevel] to pass arguments
-  /// https://docs.ably.com/client-lib-development-guide/features/#TO3b
+  /// Controls the verbosity of the logs output from the library. Levels include
+  /// `verbose`, `debug`, `info`, `warn` and `error`.
   LogLevel logLevel = LogLevel.info;
 
-  /// for development environments only
-  ///
-  /// https://docs.ably.com/client-lib-development-guide/features/#TO3k2
+  /// Enables a non-default Ably host to be specified. For development
+  /// environments only.
   String? restHost;
```

Comment:
As discussed on slack - since those default values aren't assigned in Flutter, I removed mentions of them from the docstrings. There should be an issue created for adding those default values, docstrings should imo be updated in the PR that will do that work.